cql3/expr/expression.cc

```diff
@@ -1013,6 +1013,14 @@ bool is_supported_by(const expression& expr, const secondary_index::index& idx)
                             return false;
                         },
                         [&] (const token&) { return false; },
+                        [&] (const subscript& s) -> bool {
+                            // FIXME: This seems wrong, should probably just return false.
+                            // We don't support indexes on map entries yet.
+                            // For now behaviour copied from column_value& handler to preserve behaviour.
+                            // Fill be fixed soon in a separate patch.
+                            const column_value& col = get_subscripted_column(s);
+                            return idx.supports_expression(*col.col, oper.op);
+                        },
```

Comment:
I think this is a regression introduced by expressions. Please file an issue (and find the breaking commit).